### PR TITLE
 python.pkgs.scikitlearn: 0.21.3 -> 0.22.2.post1 

### PR DESCRIPTION
**Pull request by timokau**
###### Motivation for this change

Surprisingly to me, `scikitlearn` hasn't been updated since November. Since I couldn't find a reason for that, here's the update.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by prusnak — APPROVED** [review]

**Comment by timokau** [comment]
`nix-review`:

```
15 packages failed to build:
python37Packages.imbalanced-learn python37Packages.optuna python37Packages.persim python37Packages.pygbm python37Packages.qasm2image python37Packages.qiskit python37Packages.qiskit-aqua python37Packages.ripser python37Packages.scikit-tda python38Packages.imbalanced-learn python38Packages.persim python38Packages.pygbm python38Packages.qiskit-aqua python38Packages.ripser python38Packages.scikit-tda

53 packages built:
python37Packages.batchgenerators python37Packages.bayesian-optimization python37Packages.dask-glm python37Packages.dask-ml python37Packages.dask-xgboost python37Packages.dftfit python37Packages.graspy python37Packages.hdbscan python37Packages.hmmlearn python37Packages.ignite python37Packages.kmapper python37Packages.librosa python37Packages.lightgbm python37Packages.mlrose python37Packages.nilearn python37Packages.osmnx python37Packages.qiskit-ignis python37Packages.scikit-bio python37Packages.scikit-optimize python37Packages.scikitlearn python37Packages.sklearn-deap python37Packages.textacy python37Packages.umap-learn python37Packages.vowpalwabbit python37Packages.word2vec python37Packages.xgboost python38Packages.batchgenerators python38Packages.bayesian-optimization python38Packages.dask-glm python38Packages.dask-ml python38Packages.dask-xgboost python38Packages.dftfit python38Packages.graspy python38Packages.hdbscan python38Packages.hmmlearn python38Packages.ignite python38Packages.kmapper python38Packages.librosa python38Packages.lightgbm python38Packages.mlrose python38Packages.nilearn python38Packages.osmnx python38Packages.qiskit-ignis python38Packages.scikit-bio python38Packages.scikit-optimize python38Packages.scikitlearn python38Packages.sklearn-deap python38Packages.textacy python38Packages.umap-learn python38Packages.vowpalwabbit python38Packages.word2vec python38Packages.xgboost tebreak
```

**Comment by timokau** [comment]
All of the failures are broken on master already.

File: pkgs/development/python-modules/scikitlearn/default.nix [hunk 1]
```diff
@@ -61,6 +52,12 @@ buildPythonPackage rec {
 
   meta = with stdenv.lib; {
     description = "A set of python modules for machine learning and data mining";
+    changelog = let
+      major = versions.major version;
+      minor = versions.minor version;
+      dashVer = replaceChars ["."] ["-"] version;
+    in
+      "https://scikit-learn.org/stable/whats_new/v${major}.${minor}.html#version-${dashVer}";
```
Comment (bhipple) [inline pkgs/development/python-modules/scikitlearn/default.nix:60]: This is great.  I'm going to update the `nixpkgs-update` bot to drop a link to `meta.description` in the PR description if it exists, btw:

https://github.com/ryantm/nixpkgs-update/pull/200

I spent a lot of time looking for release notes when reviewing PRs, so it'll be good to have it a little more streamlined!

**Review by bhipple — APPROVED** [review]
Actual bump looks good.  :+1:

**Review by jonringer — APPROVED** [review]

**Comment by jonringer** [comment]
@GrahamcOfBorg build python3Packages.scikitlearn
@GrahamcOfBorg build python38Packages.scikitlearn

**Pull request merged**